src/main/java/domain/entities/core/EventLoop.java

```diff
@@ -0,0 +1,78 @@
+package domain.entities.core;
+
+import core.exceptions.DisposableException;
+
+/**
+ * This shall serve as the base for implementing an event loop. An event loop
+ * loops forever unless the executable in it returns ExitStatus.terminate.
+ * Also, when the loop terminates, it will call the dispose method, which
+ * will dispose the things that are registered to the disposables.
+ */
+public abstract class EventLoop implements Disposable {
+    /**
+     * The root executable that shall be executed in the loop.
+     */
+    private final Commandable rootCommandable;
+    /**
+     * The StringReadable that shall be used for providing the next line of
+     * input to the event loop.
+     */
+    private final StringReadable reader;
+    /**
+     * The writer for writing the error messages.
+     */
+    private final Writable errorWriter;
+
+    public EventLoop(Commandable rootCommandable, StringReadable reader,
+                     Writable errorWriter) {
```

Comment:
I notice that many of your constructors start the next line at more than 8 spaces from the parent line. I couldn't find a rule regarding constructors but the standard does indeed say the next line should start 8 spaces after the parent line. 

Perhaps this could warrant another look?